mongoimport/mongoimport.go

```diff
@@ -487,20 +487,19 @@ func (imp *MongoImport) runInsertionWorker(readDocs chan bson.D) (err error) {
 	if err != nil {
 		return fmt.Errorf("error connecting to mongod: %v", err)
 	}
-	defer session.Close()
-	if err = imp.configureSession(session); err != nil {
-		return fmt.Errorf("error configuring session: %v", err)
-	}
-	collection := session.DB(imp.ToolOptions.DB).C(imp.ToolOptions.Collection)
+	collection := session.Database(imp.ToolOptions.DB).Collection(imp.ToolOptions.Collection)
 
 	var inserter flushInserter
 	if imp.IngestOptions.Mode == modeInsert {
-		inserter = db.NewBufferedBulkInserter(collection, imp.IngestOptions.BulkBufferSize, !imp.IngestOptions.StopOnError)
+		bufferedInserter := db.NewBufferedBulkInserter(collection, imp.IngestOptions.BulkBufferSize, !imp.IngestOptions.StopOnError)
+		bufferedInserter.SetBypassDocumentValidation(imp.IngestOptions.BypassDocumentValidation)
```

Comment:
I don't see us setting bypassDocumentValidation before. were we just ignoring the option?

Reply:
The only code I can see about document validation is https://github.com/mongodb/mongo-tools/blob/master/mongoimport/main/mongoimport.go#L63. I think mgo allowed setting this at a session (client?) level but the SetBypassDocumentValidation code now just panics so I'm not sure if this was actually being used somewhere.

Reply:
yeah, an mgo "session" is a client. where exactly is the panic happening?

Reply:
the new common code panics when SetBypassDocumentValidation is called (https://github.com/mongodb/mongo-tools-common/blob/master/db/db.go#L147) because this isn't an option that can be set at the client level anymore.

Reply:
gotcha, so we set it for the operation now. makes sense 👍